### PR TITLE
Update daemonsets for sysdig agent to use new api version [SMAGENT-2004]

### PR DESCRIPTION
**Pull request by papajulio**
Old one has been deprecated in 1.16 kubernetes version

### CODE REVIEW

File: agent_deploy/kubernetes/sysdig-agent-daemonset-v1.yaml [hunk 1]
```diff
@@ -3,13 +3,17 @@
 # We recommend to use v2 daemonset which instead leverages Kubernetes
 # best practices like Secrets and ConfigMaps
 ##########################################################################
-apiVersion: extensions/v1beta1
+# apiVersion: extensions/v1beta1  # If you are in Kubernetes version 1.8 or less please use this line instead of the following one
```
Comment (anoop-sysd) [inline agent_deploy/kubernetes/sysdig-agent-daemonset-v1.yaml]: lets not bother updating v1. we want to get rid of it at some point.

File: agent_deploy/kubernetes/sysdig-agent-daemonset-v1.yaml [hunk 2]
```diff
@@ -3,13 +3,17 @@
...
+apiVersion: apps/v1
 kind: DaemonSet
 metadata:
   name: sysdig-agent
   labels:
     app: sysdig-agent
 spec:
+  selector:
```
Comment (anoop-sysd) [inline agent_deploy/kubernetes/sysdig-agent-daemonset-v1.yaml]: this change seems unrelated to what the PR states. and btw, why is this required? it hasn't been so far afaik.

Comment (papajulio) [reply agent_deploy/kubernetes/sysdig-agent-daemonset-v1.yaml]: Because the new api require to have a selector for all objects using it. You can not change the api and that's all, at least for deployments and daemonsets

Comment (papajulio) [reply agent_deploy/kubernetes/sysdig-agent-daemonset-v1.yaml]: Error you get if you try to change only the api and don't add a selector: 

# * : Invalid value: "The edited file failed validation": ValidationError(DaemonSet.spec): missing required field "selector" in io.k8s.api.apps.v1.DaemonSetSpec

**Comment by anoop-sysd** [comment]
@miketnt , just a heads-up, the agent needs to be updated as well which is work-in-progress. See SMPROD-2283.
